src/inmanta/deploy/work.py

```diff
@@ -345,7 +347,22 @@ def deploy_with_context(
         for resourceId in queued:
             if resourceId in resources:
                 task = tasks.Deploy(resource=resourceId)
-                self.agent_queues._in_progress[task] = min(self.agent_queues._in_progress[task], priority)
+                assert task in self.agent_queues._in_progress
+                actual_task = [e for e in self.agent_queues._in_progress.keys() if e == task][0]
+                if self.agent_queues._in_progress[task] > priority:
+                    self.agent_queues._in_progress[task] = priority
+                    actual_task.set_reason(reason=reason)
+                elif self.agent_queues._in_progress[task] < priority or (
+                    self.agent_queues._in_progress[task] == priority and actual_task.reason == reason
+                ):
+                    # We don't need to do anything in that particular case, the lowest priority is already set
+                    pass
+                else:
+                    raise RuntimeError(
+                        f"Current task is equals to 2 different priorities: currently stored `{actual_task}` "
+                        f"- `{self.agent_queues._in_progress[task]}` | new priority `{reason}` - "
+                        f"`{priority}"
+                    )
```

Comment:
I don't think this is necessary (especially if you add the resource id in the reason)

Reply:
I'm not sure to understand your comment

Reply:
> What do you feel about also adding the resource id of the resource that triggers the task? 

Related to this. If we add the resource id of the resource that triggers the task we might get into a situation where this would change and we would be fine with it.

Reply:
The wording is also a bit weird

Reply:
But overall, no strong feelings regarding this